Error while checking build/O2/o2-cs8 for 45d405a42da3efdfc6910c7ae8ab3ebf1aaff551 at 2021-11-03 09:56:
```
sw/BUILD/O2-latest/log
FAILED: Detectors/AOD/CMakeFiles/O2lib-AODProducerWorkflow.dir/src/AODProducerWorkflowSpec.cxx.o 
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CONTAINER_DYN_LINK -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SERIALIZATION_DYN_LINK -DDPL_ENABLE_BACKTRACE -DENABLE_UPGRADES -DFAIRMQ_HAS_STD_FILESYSTEM=1 -DFMT_LOCALE -DO2lib_AODProducerWorkflow_EXPORTS -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/AOD/include -IDetectors/AOD -IDetectors/AOD/src -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/Vertexing/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Common/Utils/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/common/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Common/Constants/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/Common -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Framework/Logger/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Common/MathUtils/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Headers/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/MemoryResources/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/Utils -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/Common/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Utilities/rANS/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Parameters/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Common/Types/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/TPC/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/simulation/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/Base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Common/Field/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Framework/Core/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/CCDB/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Framework/Foundation/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Framework/Foundation/3rdparty/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Utilities/PCG/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Common/SimConfig/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Algorithm/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/ITSMFT/ITS/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/ITSMFT/common/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/common/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/Raw/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FT0/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FT0/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FT0/simulation/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/FIT/FT0/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/Calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/FIT/common/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FT0/calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/GlobalTracking/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/ITSMFT/MFT/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/MFT/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/common/simulation/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/common/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Framework/Utils/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/MUON/MCH/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/MUON/MID/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/FIT/FV0/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FV0/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/FIT/FDD/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FDD/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/ZDC/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ZDC/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/TOF/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/TRD/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/CTP/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/CPV/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/CPV/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/PHOS/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/PHOS/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/DataFormats/Detectors/EMCAL/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Definitions -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/DataTypes -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/common/calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FV0/calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/GlobalTracking/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/ITS/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/ITS/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/TPCFastTransformation -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/TPCFastTransformation/devtools -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Interface -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TOF/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TOF/calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/DCS/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TOF/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TOF/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TOF/workflowIO/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/MFT/tracking/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Tracking/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Steer/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FT0/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FT0/raw/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/raw/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FDD/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FDD/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FDD/raw/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FV0/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FV0/reconstruction/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FV0/simulation/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/FIT/FV0/raw/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/GlobalTrackingWorkflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/GlobalTrackingWorkflow/readers/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/GlobalTrackingWorkflow/helpers/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/ITS/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/common/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Base -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/SliceTracker -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/TPCConvert -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/dEdx -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/ITS -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/TRDTracking -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/qa -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/display -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Global -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/HLTHeaders -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Merger -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Refit -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/Debug -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/DataCompression -I/sw/SOURCES/O2/7497-slc8_x86-64/0/GPU/GPUTracking/TPCClusterFinder -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TRD/base/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/ITSMFT/MFT/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/calibration/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/simulation/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/spacecharge/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/qc/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/workflow/readers/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/TPC/monitor/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Workflow/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/CTF/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Clustering/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Mapping/Interface/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/PreClustering/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/TimeClustering/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Raw/Common/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Raw/Decoder/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Raw/ElecMap/include -I/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/MUON/MCH/Raw/ImplHelpers -isystem /sw/slc8_x86-64/ROOT/v6-24-06-18/include -isystem /sw/slc8_x86-64/boost/v1.75.0-19/include -isystem /sw/slc8_x86-64/fmt/7.1.0-13/include -isystem /sw/slc8_x86-64/FairLogger/v1.10.1-5/include -isystem /sw/slc8_x86-64/FairRoot/v18.4.2-local2/include -isystem /sw/slc8_x86-64/arrow/v5.0.0-alice3-1/include -isystem /sw/slc8_x86-64/Vc/1.4.1-13/include -isystem /sw/slc8_x86-64/ms_gsl/3.1.0-7/include -isystem /sw/slc8_x86-64/FairMQ/v1.4.42-14/include/fairmq -isystem /sw/slc8_x86-64/FairMQ/v1.4.42-14/include -isystem /sw/slc8_x86-64/Common-O2/v1.6.0-19/include -isystem /sw/slc8_x86-64/Configuration/v2.6.3-7/include -isystem /sw/slc8_x86-64/libInfoLogger/v2.4.1-1/include -isystem /sw/slc8_x86-64/Monitoring/v3.8.11-6/include -isystem /sw/slc8_x86-64/libjalienO2/0.1.3-10/include -isystem /sw/slc8_x86-64/RapidJSON/v1.1.0-alice2-13/include -isystem /sw/slc8_x86-64/libuv/v1.40.0-13/include -isystem /sw/slc8_x86-64/DebugGUI/v0.6.1-8/include -isystem /sw/slc8_x86-64/GLFW/3.3.2-13/include -fPIC -O2 -std=c++17 -Werror -Wno-error=deprecated-declarations -O2 -g -fPIC -std=gnu++17 -MD -MT Detectors/AOD/CMakeFiles/O2lib-AODProducerWorkflow.dir/src/AODProducerWorkflowSpec.cxx.o -MF Detectors/AOD/CMakeFiles/O2lib-AODProducerWorkflow.dir/src/AODProducerWorkflowSpec.cxx.o.d -o Detectors/AOD/CMakeFiles/O2lib-AODProducerWorkflow.dir/src/AODProducerWorkflowSpec.cxx.o -c /sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx: In member function 'virtual void o2::aodproducer::AODProducerWorkflowDPL::run(o2::framework::ProcessingContext&)':
/sw/SOURCES/O2/7497-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:988:114: error: request for member 'getEventRecords' in 'mcReader.std::unique_ptr<o2::steer::MCKinematicsReader>::operator->()->o2::steer::MCKinematicsReader::getDigitizationContext()', which is of pointer type 'const o2::steer::DigitizationContext*' (maybe you meant to use '->' ?)
  988 |   collectBCs(fddRecPoints, ft0RecPoints, fv0RecPoints, primVertices, mUseMC ? mcReader->getDigitizationContext().getEventRecords() : std::vector<o2::InteractionTimeRecord>{}, bcsMap);
      |                                                                                                                  ^~~~~~~~~~~~~~~
[3369/3381] Building CXX object Steer/DigitizerWorkflow/CMakeFiles/O2exe-sim-digitizer-workflow.dir/src/ZDCDigitizerSpec.cxx.o
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/7497/45d405a42da3efdfc6910c7ae8ab3ebf1aaff551/build_O2_o2-cs8/pretty.html).

--- a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
+++ b/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
@@ -974,17 +974,15 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
   auto v0sCursor = v0sBuilder.cursor<o2::aod::StoredV0s>();
   auto zdcCursor = zdcBuilder.cursor<o2::aod::Zdcs>();
 
-  o2::steer::MCKinematicsReader mcReader("collisioncontext.root");
-  const auto mcContext = mcReader.getDigitizationContext();
-  const auto& mcRecords = mcContext->getEventRecords();
-  const auto& mcParts = mcContext->getEventParts();
-
-  LOG(DEBUG) << "FOUND " << mcRecords.size() << " records";
-  LOG(DEBUG) << "FOUND " << mcParts.size() << " parts";
+  std::unique_ptr<o2::steer::MCKinematicsReader> mcReader;
+  if (mUseMC) {
+    mcReader = std::make_unique<o2::steer::MCKinematicsReader>("collisioncontext.root");
+    LOG(DEBUG) << "FOUND " << mcReader->getDigitizationContext()->getEventRecords().size()
+               << " records" << mcReader->getDigitizationContext()->getEventParts().size() << " parts";
+  }
 
   std::map<uint64_t, int> bcsMap;
-  collectBCs(fddRecPoints, ft0RecPoints, fv0RecPoints, primVertices, mcRecords, bcsMap);
-
+  collectBCs(fddRecPoints, ft0RecPoints, fv0RecPoints, primVertices, mUseMC ? mcReader->getDigitizationContext()->getEventRecords() : std::vector<o2::InteractionTimeRecord>{}, bcsMap);
   const auto* dh = o2::header::get<o2::header::DataHeader*>(pc.inputs().getFirstValid(true).header);
   o2::InteractionRecord startIR = {0, dh->firstTForbit};
 
@@ -1071,7 +1069,9 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
     // TODO: figure out collision weight
     float mcColWeight = 1.;
     // filling mcCollision table
-    int nMCCollisions = mcContext->getNCollisions();
+    int nMCCollisions = mcReader->getDigitizationContext()->getNCollisions();
+    const auto& mcRecords = mcReader->getDigitizationContext()->getEventRecords();
+    const auto& mcParts = mcReader->getDigitizationContext()->getEventParts();
     for (int iCol = 0; iCol < nMCCollisions; iCol++) {
       auto time = mcRecords[iCol].getTimeNS();
       auto globalBC = mcRecords[iCol].toLong();
@@ -1090,7 +1090,7 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
           // FIXME:
           // use generators' names for generatorIDs (?)
           short generatorID = sourceID;
-          auto& header = mcReader.getMCEventHeader(sourceID, eventID);
+          auto& header = mcReader->getMCEventHeader(sourceID, eventID);
           mcCollisionsCursor(0,
                              bcID,
                              generatorID,
@@ -1294,7 +1294,7 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
 
   if (mUseMC) {
     // filling mc particles table
-    fillMCParticlesTable(mcReader,
+    fillMCParticlesTable(*mcReader,
                          mcParticlesCursor,
                          primVer2TRefs,
                          primVerGIs,

